### DISCUSSION TITLE
[Scenario Editor] Allow event triggers to exist across multiple spaces

### DISCUSSION THREAD

**Discussion by Granfalegion** [ihhub/fheroes2#9342] [Ideas]
In conversations with map authors, I've heard some frustration about having to produce intricate and complex terrains in order to force a single-square bottleneck to host an event trigger.  While there's a lot to enjoy about maps that can produce these clever interaction points, it seems like an easy answer to mitigating these frustrations would be to allow events to have multiple triggering locations.  This would cleanly allow authors to define, say, a boundary of squares that can trigger an event rather than a single bottleneck.  This opens up more story-driven design choices without forcing map constructions that authors might wish to avoid.

**Comment by ihhub** [comment]
Hi @Granfalegion , do you mean the same concept as in HoMM4 where events are bound to an event name?

**Reply by Granfalegion** [reply]
I'm unfamiliar with H4 and its features, but I imagine that an implementation here would involve some kind of central identifier that the different event spaces would share so their triggering could be tracked collectively.  Such an identifier may well be an event name.

**Reply by indrek-l** [reply]
A bit late to the discussion but yes, it would be great if the placed events followed the logic of HoMM4 where the event object on the map and the event definition are decoupled. Like in HoMM4, the map event object specs could have a dropdown menu for choosing an event ID. The event definition (ID, text, rewards, conditions) could be specified under "Events" in Map Specs which currently only holds timed events. The "Events" button could open a panel where one could select e.g. "Timed Events" or "Placed Events". This would also serve as a set-up for additional event types in the future, e.g. "Town Capture Events".

**Comment by Branikolog** [comment]
That's a good one. Indeed, allowing to trigger the same event for the area around it (or combine several events into a new triggering situation) could bring more freedom to map makers.
